operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +68,16 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type BootImageUpdateModeEnum string
+
+const (
+	// Enabled describes an operator that is managing the boot images of all machine sets
+	Enabled BootImageUpdateModeEnum = "Enabled"
+	// Selected describes an operator that is managing the boot images of the selected machine sets, listed in the secondary field BootImageUpdateEnrolledMachineSets
+	Selected BootImageUpdateModeEnum = "Selected"
+	// Disabled describes an operator that is not managing boot images
+	Disabled BootImageUpdateModeEnum = "Disabled"
+)
```

Comment:
Looking at this, I think you want a [discriminated union](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions)